include/seqan3/alphabet/structure/dot_bracket3.hpp

```diff
@@ -0,0 +1,324 @@
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \ingroup structure
+ * \author Joerg Winkler <j.winkler AT fu-berlin.de>
+ * \brief Contains the dot bracket format for RNA structure.
+ */
+
+#pragma once
+
+#include <cassert>
+#include <string>
+#include <vector>
+
+#include <seqan3/alphabet/structure/concept.hpp>
+
+// ------------------------------------------------------------------
+// db3
+// ------------------------------------------------------------------
+
+namespace seqan3
+{
+
+/*!\brief The three letter RNA structure alphabet of the characters ".()".
+ * \ingroup structure
+ *
+ * \details
+ * The brackets denote RNA base pair interactions. Every left bracket (BL) must have a corresponding right bracket (BR).
+ * Pseudoknots cannot be expressed in this format. A dot (.) represents a character that is not paired (NP).
+ *
+ *~~~~~~~~~~~~~~~{.cpp}
+ * GCGGAUUUAGCUCAGUUGGGAGAGCGCCAGACUGAAGAUUUGGAGGUCCUGUGUUCGAUCCACAGAAUUCGCA
+ * (((((((..((((........)))).((((.........)))).....(((((.......)))))))))))).
+ *~~~~~~~~~~~~~~~
+ */
+
+struct db3
+{
+    //!\brief The type of the alphabet when converted to char (e.g. via to_char()).
+    using char_type = char;
+    //!\brief The type of the alphabet when represented as a number (e.g. via to_rank()).
+    using rank_type = uint8_t;
+
+    /*!\name Letter values
+     * \brief Static member "letters" that can be assigned to the alphabet or used in aggregate initialization.
+     * \details Similar to an Enum interface. *Don't worry about the `internal_type`.*
+     */
+    //!\{
+    static const db3 NP; // not paired
+    static const db3 BL; // bracket left
+    static const db3 BR; // bracket right
+    static const db3 NA; // not available
+    //!\}
+
+    /*!\name Read functions
+     * \{
+     */
+    //!\brief Return the letter as a character of char_type.
+    constexpr char_type to_char() const noexcept
+    {
+        return value_to_char[static_cast<rank_type>(_value)];
+    }
+
+    //!\brief Return the letter's numeric value or rank in the alphabet.
+    constexpr rank_type to_rank() const noexcept
+    {
+        return static_cast<rank_type>(_value);
+    }
+    //!\}
+
+    /*!\name Write functions
+     * \{
+     */
+    //!\brief Assign from a character.
+    constexpr db3 & assign_char(char_type const c) noexcept
```

Comment:
weren't you the one that didn't like one-character-names ? :open_mouth: 
Also I think you should document parameters and return values of public functions (even if there are obvious).